pkg/controller/common/certificates/reconcile.go

```diff
@@ -140,17 +139,7 @@ func (r *Reconciler) removeCAAndHTTPCertsSecrets() error {
 }
 
 func deleteIfExists(c k8s.Client, secretRef types.NamespacedName) error {
-	var secret corev1.Secret
-	err := c.Get(context.Background(), secretRef, &secret)
-	if err != nil && apierrors.IsNotFound(err) {
-		return nil
-	} else if err != nil {
-		return err
-	}
-	log.Info("Deleting secret", "namespace", secretRef.Namespace, "secret_name", secretRef.Name)
-	err = c.Delete(context.Background(), &secret)
-	if err != nil && apierrors.IsNotFound(err) {
-		return nil
-	}
-	return err
+	return k8s.DeleteSecretIfExists(c, secretRef, func() {
+		log.Info("Deleting secret", "namespace", secretRef.Namespace, "secret_name", secretRef.Name)
```

Comment:
I'm a little perplexed about the usefulness of this log. If we do without it, we can delete `deleteIfExists` and use `k8s.DeleteSecretIfExists` instead.